dll/win32/shell32/CDefView.cpp

```diff
@@ -3182,6 +3412,18 @@ HRESULT WINAPI CDefView::GetItemObject(UINT uItem, REFIID riid, LPVOID *ppvOut)
     return hr;
 }
 
+FOLDERVIEWMODE CDefView::GetDefaultViewMode()
+{
+    FOLDERVIEWMODE mode = ((m_FolderSettings.fFlags & FWF_DESKTOP) || !IsOS(OS_SERVERADMINUI)) ? FVM_ICON : FVM_DETAILS;
+    FOLDERVIEWMODE temp;
+    if (SUCCEEDED(_DoFolderViewCB(SFVM_DEFVIEWMODE, 0, (LPARAM)&temp)))
+    {
+        if (temp >= FVM_FIRST && temp <= FVM_LAST)
+            mode = temp;
+    }
```

Comment:
Nested if-s may be collapsed
```suggestion
    if (SUCCEEDED(_DoFolderViewCB(SFVM_DEFVIEWMODE, 0, (LPARAM)&temp)) &&
        (temp >= FVM_FIRST && temp <= FVM_LAST))
    {
        mode = temp;
    }
```